pkg/ocm/sca/sca.go

```diff
@@ -185,41 +185,56 @@ func (c *Controller) createSecret(ocmData *Response) (*v1.Secret, error) {
 		},
 		Type: v1.SecretTypeOpaque,
 	}
-	cm, err := c.coreClient.Secrets(targetNamespaceName).Create(c.ctx, newSCA, metav1.CreateOptions{})
+	cm, err := c.coreClient.Secrets(targetNamespaceName).Create(ctx, newSCA, metav1.CreateOptions{})
 	if err != nil {
 		return nil, err
 	}
 	return cm, nil
 }
 
 // updateSecret updates provided secret with given data
-func (c *Controller) updateSecret(s *v1.Secret, ocmData *Response) (*v1.Secret, error) {
+func (c *Controller) updateSecret(ctx context.Context, s *v1.Secret, ocmData *Response) (*v1.Secret, error) {
 	s.Data = map[string][]byte{
 		entitlementAttrName:    []byte(ocmData.Cert),
 		entitlementKeyAttrName: []byte(ocmData.Key),
 	}
-	s, err := c.coreClient.Secrets(s.Namespace).Update(c.ctx, s, metav1.UpdateOptions{})
+	s, err := c.coreClient.Secrets(s.Namespace).Update(ctx, s, metav1.UpdateOptions{})
 	if err != nil {
 		return nil, err
 	}
 	return s, nil
 }
 
+// getArch check the value of GOARCH and return a valid representation for
+// OCM certificates API
+func getArch() string {
+	validArchs := map[string]string{
+		"amd64": "x86_64",
+		"i386":  "x86",
```

Comment:
So the remaining archs don't require additional mapping right?

Reply:
Yes, I tried the endpoint with other supported architectures and they don't need any translation mapping like these two.